packages/mobile/src/home/RecentlyUsedDapps.tsx

```diff
@@ -0,0 +1,137 @@
+import { Colors } from '@celo/react-components/styles/colors'
+import fontStyles from '@celo/react-components/styles/fonts'
+import { Spacing } from '@celo/react-components/styles/styles'
+import variables from '@celo/react-components/styles/variables'
+import * as React from 'react'
+import { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import {
+  Image,
+  NativeScrollEvent,
+  ScrollView,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+} from 'react-native'
+import { recentDappsSelector } from 'src/app/selectors'
+import { Dapp } from 'src/app/types'
+import ProgressArrow from 'src/icons/ProgressArrow'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import useSelector from 'src/redux/useSelector'
+
+interface Props {
+  onSelectDapp(dapp: Dapp): void
+}
+
+const DAPP_ICON_SIZE = 68
+
+function RecentlyUsedDapps({ onSelectDapp }: Props) {
+  const recentlyUsedDapps = useSelector(recentDappsSelector)
+  const [currentIndex, setCurrentIndex] = useState(0)
+  const { t } = useTranslation()
+
+  const handleScroll = (event: { nativeEvent: NativeScrollEvent }) => {
+    const nextIndex = Math.round(event.nativeEvent.contentOffset.x / variables.width)
+    if (nextIndex === currentIndex) {
+      return
+    }
+
+    setCurrentIndex(nextIndex)
+  }
+
+  const onPressAllDapps = () => {
+    navigate(Screens.DAppsExplorerScreen)
```

Comment:
Can you check with Gowtham if we want any analytics for this (or other parts)?

Reply:
yes good point! there's a separate analytics [ticket](https://app.zenhub.com/workspaces/retention-squad-sprint-board-612f2ef98ef2b800116e6043/issues/valora-inc/wallet/1965) actually, i'll put the changes in a separate PR if you're okay with that